src/react-apps/ux-editor/src/containers/FormFiller.tsx

```diff
@@ -96,15 +99,25 @@ export class FormFillerComponent extends React.Component<IFormFillerProps, IForm
   }
 }
 
-const mapStateToProps = (state: IAppState, empty: any): IFormFillerProps => {
-  return {
-    validationErrors: state.formFiller.validationErrors,
-    unsavedChanges: state.formFiller.unsavedChanges,
-    connections: state.serviceConfigurations.APIs.connections,
-    dataModelElements: state.appData.dataModel.model,
-    designMode: state.appData.appConfig.designMode,
-    formDataCount: Object.keys(state.formFiller.formData).length,
+const makeMapStateToProps = () => {
+  const GetFormDataCount = makeGetFormDataCountSelector();
+  const GetDesignMode = makeGetDesignModeSelector();
+  const GetDataModel = makeGetDataModelSelector();
+  const GetApiConnections = makeGetApiConnectionsSelector();
+  const GetUnsavedChanges = makeGetUnsavedChangesSelector();
+  const GetValidationErrors = makeGetValidationErrorsSelector();
```

Comment:
I see that multiple places we use the same methods like `makeGetFormDataCountSelctor` and `makeGetDataModelSelectors`. Any possiblity to have these run just one place?

Reply:
They are not being used in the same `mapStateToProps`, are they? Having them run in just one place would be like selecting from the state (what we did before) in just one place, then we would have to pass everything down through props instead.

Reply:
It's the same `makeGetDataModelSelector`-function.
You could export a const that has the value of `makeGetDataModelSelector()` from the selector file.

Reply:
Unfortunately this does not work when we have multiple component instances sharing the same selector. Each instance needs to have its own private copy of the selector.

Reply:
Allright 👍